test-lib.sh

```diff
@@ -60,39 +62,12 @@ function ct_init() {
 # Uses: $EXPECTED_EXIT_CODE - expected container exit code
 # Uses: $TESTSUITE_RESULT - overall result of all tests
 function ct_cleanup() {
-  ct_show_resources
-  ct_show_results
+  echo "$LINE"
+  echo "Cleaning of testing containers and images started."
+  echo "It may take a few seconds."
+  echo "$LINE"
   ct_clean_app_images
-
-  if [[ -z ${CID_FILE_DIR:-} ]]; then
-    echo "The \$CID_FILE_DIR is not set. Container cleaning is to be skipped."
-    exit "${TESTSUITE_RESULT:-0}"
-  fi;
-
-  echo "Examining CID files in \$CID_FILE_DIR=$CID_FILE_DIR"
-  for cid_file in "$CID_FILE_DIR"/* ; do
-    [ -f "$cid_file" ] || continue
-    local container
-    container=$(cat "$cid_file")
-
-    ct_container_exists "$container" || continue
-
-    echo "Stopping and removing container $container..."
-    if ct_container_running "$container"; then
-      docker stop "$container"
-    fi
-
-    exit_status=$(docker inspect -f '{{.State.ExitCode}}' "$container")
-    if [ "$exit_status" != "$EXPECTED_EXIT_CODE" ]; then
-      echo "Dumping logs for $container"
-      docker logs "$container"
-    fi
-    docker rm -v "$container"
-    rm -f "$cid_file"
-  done
-
-  rm -rf "$CID_FILE_DIR"
-  exit "${TESTSUITE_RESULT:-0}"
+  ct_clean_containers
 }
 
 # ct_build_image_and_parse_id
```

Comment:
Personally, this function becomes complicated from my point of view. As I am not a bash expert when the build took a lot of time? It would be nice to have a description or explanation of what the code does. Even for basic users.

Reply:
The problem appeared with ruby. One of the build steps got cycled and the build never ended, the CI timeouted.
There weren't any logs too, as they are printed just after the image is build.

But sure, I can add some explanatory comments.